lib/protobuf/encoder.ex

```diff
@@ -161,8 +161,23 @@ defmodule Protobuf.Encoder do
           message:
             "struct #{inspect(other_mod)} can't be encoded as #{inspect(mod)}: #{inspect(struct)}"
 
-      _ ->
+      enumerable when is_map(enumerable) or is_list(enumerable) ->
+        IO.warn("""
+        Implicitly casting from #{inspect(enumerable)} to #{inspect(mod)} message.
+        This automatic coercion is deprecated in Protobuf 0.15 and will raise an error in future versions.
+
+        Instead of:
+          %Parent{child: %{name: ""}}
+
+        Build child structs explicitly:
+          %Parent{child: %Child{name: ""}}
+        """)
+
         do_encode_to_iodata(struct(mod, msg))
+
+      other ->
+        raise Protobuf.EncodeError,
+          message: "#{inspect(other)} is invalid for type #{inspect(mod)}"
```

Comment:
This would print weird for long structs. Suggestion:

```suggestion
          message: "invalid value for type #{inspect(mod)}: #{inspect(other)}"
```